addon/-private/system/store.js

```diff
@@ -921,6 +925,9 @@ Store = Service.extend({
       // will once again convert the records to snapshots for adapter.findMany()
       let snapshots = new Array(totalItems);
       for (let i = 0; i < totalItems; i++) {
+        if (internalModels[i].hasScheduledDestroy()) {
+          internalModels[i].cancelDestroy();
```

Comment:
Why are we canceling in two places? Does the test catch both instances where canceling is required?

Reply:
per OOB discussion, moving all cancelation to flush